File: WillMoveToOwnRepo/ProjectorRays/src/ProjectorRays.DotNet/director/Chunks/RaysScriptContextChunk.cs
using ProjectorRays.Common;
using ProjectorRays.Director;
using ProjectorRays.LingoDec;

namespace ProjectorRays.director.Chunks;

public class RaysScriptContextChunk : RaysChunk
{
    public ScriptContext Context;

    public RaysScriptContextChunk(RaysDirectorFile? dir) : base(dir, ChunkType.ScriptContextChunk)
    {
        Context = new ScriptContext(dir?.Version ?? 0, dir!);
    }

    public override void Read(ReadStream stream)
    {
        Context.Read(stream);
    }

    public override void WriteJSON(RaysJSONWriter json)
    {
        json.StartObject();
        json.WriteField("entryCount", Context.EntryCount);
        json.WriteField("flags", Context.Flags);
        json.EndObject();
    }
}

File: WillMoveToOwnRepo/ProjectorRays/src/ProjectorRays_Python/director/chunks/rays_script_context_chunk.py
from .rays_chunk import RaysChunk, ChunkType
from ...common.stream import ReadStream

class RaysScriptContextChunk(RaysChunk):
    def __init__(self, dir=None):
        super().__init__(dir, ChunkType.ScriptContextChunk)
        self.context_data = b""

    def read(self, stream: ReadStream):
        self.context_data = stream.read_bytes(len(stream.data) - stream.pos)
